### ISSUE TITLE
F-droid version

### ISSUE THREAD

**Issue by Monirzadeh** [yamin8000/Owl2#93]
Hi
Can we have that on F-droid?

**Comment by yamin8000** [comment]
> Hi Can we have that on F-droid?

I'm very keen to put this app on F-Droid, however, the API key is git ignored and is only available in my system.  
One solution is to change the app in a way that doesn't require a single API key and each user requests a key from Owlbot dictionary or somehow put my API key through some environmental variables when it's compiled in GitLab and F-Droid.

**Comment by IzzySoft** [comment]
@yamin8000 meanwhile, can you please make sure that the APK attached at the release matches the release name? Currently, the one at 1.3.2 identifies as 1.3.0 (and thus my updater fetches it each day just to throw it away as it's already here). Is it even the correct APK, or did you accidentally attach the wrong one?

**Comment by yamin8000** [comment]
> @yamin8000 meanwhile, can you please make sure that the APK attached at the release matches the release name? Currently, the one at 1.3.2 identifies as 1.3.0 (and thus my updater fetches it each day just to throw it away as it's already here). Is it even the correct APK, or did you accidentally attach the wrong one?

I don't have access to my PC currently so I can't check the problem but I just checked that the apk of the free one uploaded for 1.3.2 release is indeed 1.3.2.  
Source is 1.3.2 too.  
What do you exactly mean it identifies as 1.3.0?

**Comment by IzzySoft** [comment]
Haha, wrong error… Still an error. I just dug into it:

Take a closer look at the `https://github.com/yamin8000/Owl2/releases/download/1.3.2/app-free-release.apk` please, it has the wrong package name! It should be `io.github.yamin8000.owl.free`, but it is `io.github.yamin8000.owl`. Versioning is correct, but as it has the wrong package name, it's assigned to the `io.github.yamin8000.owl` (which still exists but has updates disabled, else I would have noticed due to collision), so `io.github.yamin8000.owl.free` is not updated, and on the next round the game starts over :see_no_evil:

So it's not the version that needs to be fixed, but the package name, if you please? Or is this intended (despite of explicitly changed with v1.3.0), so I shall simply replace `io.github.yamin8000.owl` instead? Would work as well.

**Comment by IzzySoft** [comment]
PS: while keeping the old package name would work as well, it might cause some confusion to those having switched as the description said so:

> NOTE: This flavor of the app will no longer be updated here. It is continued with its [free flavor](https://apt.izzysoft.de/fdroid/index/apk/io.github.yamin8000.owl.free) (same functionality), so please switch to that instead.

Decision is yours, though.

**Comment by yamin8000** [comment]
> Haha, wrong error… Still an error. I just dug into it:
> 
> Take a closer look at the `https://github.com/yamin8000/Owl2/releases/download/1.3.2/app-free-release.apk` please, it has the wrong package name! It should be `io.github.yamin8000.owl.free`, but it is `io.github.yamin8000.owl`. Versioning is correct, but as it has the wrong package name, it's assigned to the `io.github.yamin8000.owl` (which still exists but has updates disabled, else I would have noticed due to collision), so `io.github.yamin8000.owl.free` is not updated, and on the next round the game starts over 🙈
> 
> So it's not the version that needs to be fixed, but the package name, if you please? Or is this intended (despite of explicitly changed with v1.3.0), so I shall simply replace `io.github.yamin8000.owl` instead? Would work as well.

OK. Now I understand what the actual problem is. After I created flavors for my app I specified a suffix for the free version and released a new free version with that suffix and another version with ads with the old `io.github.yamin8000.owl`, however, later I decided to remove the suffix and released a version with no suffix.  
That's a problem on my side and this inconsistency created confusion for you.
So long story short you should discard the `io.github.yamin8000.owl.free` and update `io.github.yamin8000.owl` with the free version flavor which in new version has the right version and package name.

**Comment by yamin8000** [comment]
> PS: while keeping the old package name would work as well, it might cause some confusion to those having switched as the description said so:
> 
> > NOTE: This flavor of the app will no longer be updated here. It is continued with its [free flavor](https://apt.izzysoft.de/fdroid/index/apk/io.github.yamin8000.owl.free) (same functionality), so please switch to that instead.
> 
> Decision is yours, though.

I'm sorry for the mishap, but `io.github.yamin8000.owl.free` was a mistake (1.3.0) and was reverted in the next versions.

**Comment by IzzySoft** [comment]
OK, then I'll revert to the old package name here. As you never told me, I could not have known :wink:

**Comment by yamin8000** [comment]
> > Hi Can we have that on F-droid?
> 
> I'm very keen to put this app on F-Droid, however, the API key is git ignored and is only available in my system. One solution is to change the app in a way that doesn't require a single API key and each user requests a key from Owlbot dictionary or somehow put my API key through some environmental variables when it's compiled in GitLab and F-Droid.

@IzzySoft I managed to move API keys from a git ignored file to system environment variables and then `build.gradle` and then `BuildConfig` class, theoretically app can be compiled by anyone who can generate an [OwlBot](https://owlbot.info/) dictionary API key and an [API Ninja](https://api-ninjas.com/) key.

**Comment by yamin8000** [comment]
Oh! Never mind. I just realized [Owlbot](https://owlbot.info/) will be down due to copyright issues the next day.
